### PR TITLE
Add SlsProductVersions

### PR DESCRIPTION
**Pull request by uschi2000**
This change is [Reviewable](https://reviewable.io/reviews/palantir/gradle-java-distribution/163)

### CODE REVIEW

File: src/main/groovy/com/palantir/gradle/dist/SlsProductVersions.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+[a-z0-9-]*$");
+    private static final List<Pattern> ORDERABLE_VERSION = ImmutableList.of(
+            Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+-[0-9]+-g[a-f0-9]+$"),
+            Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]$"),
+            Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+-rc[0-9]+$"),
+            Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+-beta[0-9]+$")
+    );
+
+    /**
+     * Returns tree iff the given string is a valid "orderable" SLS version.
```
Comment (ash211) [inline src/main/groovy/com/palantir/gradle/dist/SlsProductVersions.java]: nit: tree -> true

File: src/main/groovy/com/palantir/gradle/dist/SlsProductVersions.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    public static boolean isOrderableVersion(String version) {
+        return Iterables.any(ORDERABLE_VERSION, pattern -> pattern.matcher(version).matches());
+    }
+
+    /**
+     * Returns tree iff the given string is a valid "non-orderable" SLS version.
```
Comment (ash211) [inline src/main/groovy/com/palantir/gradle/dist/SlsProductVersions.java]: tree -> true

File: src/main/groovy/com/palantir/gradle/dist/SlsProductVersions.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    public static boolean isNonOrderableVersion(String version) {
+        return NON_ORDERABLE_VERSION.matcher(version).matches();
+    }
+
+    /**
+     * Returns tree iff the given string is a valid "orderable" or "non-orderable" SLS version.
```
Comment (ash211) [inline src/main/groovy/com/palantir/gradle/dist/SlsProductVersions.java]: tree -> true

**Comment by uschi2000** [comment]
all trees replaced by trues.

**Comment by uschi2000** [comment]
Anything else here?

**Comment by uschi2000** [comment]
Merging so I can move on; let me know if you want changes.

**Comment by markelliot** [comment]
Looks fine to me, only thought is we might want to actually publish an sls-spec Java lib with this utility in it. Don't feel strongly, especially since this is really the only place we'd need a Java version.

**Comment by uschi2000** [comment]
Two options for such a library: could live in sls-spec repo or here. Either
we, I think we should consider renaminf GJD to sls-packaging at some point.

**Comment by markelliot** [comment]
Yeah, fine leaving it here, also fine with renaming the repo.

**Comment by uschi2000** [comment]
Made some issues, punting on both for now.

**Pull request merged**